addons/gdshell/commands/default_commands/ls.gd

```diff
@@ -0,0 +1,119 @@
+extends GDShellCommand
+
+
+signal done_with_instance(instance: Node)
+
+
+var created_an_instance:= false
+
+
+func _main(argv: Array, data) -> Dictionary:
+	done_with_instance.connect(_on_done_with_instance)
+	var path: String
+	
+	# If additional functionality is added, this should be moved to it's own function
+	# But i think i addressed most of what an ls command could do
+	var starting_node = get_starting_node(argv)
+	if starting_node is Dictionary:
+		return starting_node
+	var tree_dict = get_tree_dict(starting_node, starting_node)
+	output_tree_dict(tree_dict)
+	
+	data = tree_dict
+	return {}
```

Comment:
`data` is not returned. I is just an argument of the `_main()` function and you have to manually put it in the `Dictionary`, that gets returned.

```suggestion
	return {
		"error": 0,
		"error_string": "",
		"data": tree_dict,
	}
```